File: scripts/errors/probe-planes-api-env.mjs
#!/usr/bin/env node
/**
 * Directory maintain/probe entry for the planes-api .env preserve gate.
 *
 * Hang from directory:
 *   errors:probe:refresh
 *   errors:maintain
 * via `npm --prefix ../plane-alert run verify:planes-api-env`
 * and `npm --prefix ../plane-alert run verify:kiosk-alert`.
 */
import { spawnSync } from "node:child_process";
import path from "node:path";
import { fileURLToPath } from "node:url";

const repoRoot = path.resolve(path.dirname(fileURLToPath(import.meta.url)), "../..");
const result = spawnSync(
  "python3",
  [path.join(repoRoot, "scripts/probe-planes-api-env.py")],
  { stdio: "inherit" },
);
process.exit(result.status ?? 1);

File: scripts/probe-planes-api-env.py
#!/usr/bin/env python3
"""Maintain-timer gate: planes-api install must not clobber the Pi .env.

Fails if `scripts/pi-install-planes-api.py` would upload local functions/.env
or let `cp -a STAGING/. REMOTE_ROOT/` replace a pre-existing host .env.

OWNER: errors:probe:refresh / errors:maintain (plane-alert / planes-api-env)
PROVE: python3 scripts/pi-install-planes-api.py --self-test
       npm run verify:kiosk-alert
       npm run verify:planes-api-env
"""
from __future__ import annotations

import subprocess
import sys
from pathlib import Path

SCRIPT_DIR = Path(__file__).resolve().parent


def main() -> int:
    result = subprocess.run(
        [sys.executable, str(SCRIPT_DIR / "pi-install-planes-api.py"), "--self-test"],
        check=False,
    )
    if result.returncode == 0:
        print("OWNER: errors:probe:refresh / errors:maintain planes-api-env")
        print("PROVE: cd plane-alert && npm run verify:planes-api-env")
    return result.returncode


if __name__ == "__main__":
    raise SystemExit(main())
